File: articles/simulation-as-computational-reasoning/java/SimulationSummary.java
public class SimulationSummary {
    static double step(double x) {
        return Math.max(0.0, x + 0.08 * x - 0.03 * x - 0.04 * x);
    }
    public static void main(String[] args) {
        double stock = 100.0;
        for (int t = 0; t <= 30; t++) {
            System.out.printf("time_step=%d,stock=%.6f%n", t, stock);
            stock = step(stock);
        }
    }
}

File: articles/simulation-as-computational-reasoning/julia/simulation_summary.jl
# Simulation as Computational Reasoning - Julia scaffold
stock = 100.0
for t in 0:30
    global stock
    println((time_step=t, stock=round(stock, digits=6)))
    stock = max(0.0, stock + 0.08stock - 0.03stock - 0.04stock)
end
